presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxHistoricalData.java

```diff
@@ -27,24 +30,35 @@
 import java.util.Set;
 
 import static com.google.common.base.Preconditions.checkArgument;
-import static java.util.stream.Collectors.toSet;
+import static io.prestosql.plugin.jmx.JmxMetadata.toPattern;
+import static java.util.Locale.ENGLISH;
+import static javax.management.ObjectName.WILDCARD;
 
 public class JmxHistoricalData
 {
     private final Set<String> tables;
     private final Map<String, EvictingQueue<List<Object>>> tableData = new HashMap<>();
 
     @Inject
-    public JmxHistoricalData(JmxConnectorConfig jmxConfig)
+    public JmxHistoricalData(JmxConnectorConfig jmxConfig, MBeanServer mbeanServer)
+            throws MalformedObjectNameException
```

Comment:
I would prefer to throw PrestoException here. This exception will be burried with guice exceptions so having `PrestoException` could be a bit more user friendly.

Can you please post what kind of exception user will get during Presto initialization in case of wrong table names?

Reply:
This would be the exception thrown in case of wrong table names `javax.management.MalformedObjectNameException: Key properties cannot be empty`